coderedcms/templatetags/coderedcms_tags.py

```diff
@@ -40,6 +41,18 @@ def coderedcms_version():
 def generate_random_id():
     return ''.join(random.choice(string.ascii_letters + string.digits) for n in range(20))
 
+@register.simple_tag(takes_context=True)
+def og_image(context, page):
+    site_url = context['request'].site.root_url
+    if page.og_image:
+        relative_path = page.og_image.get_rendition('original').url
+    elif page.cover_image:
+        relative_path = page.cover_image.get_rendition('original').url
+    elif LayoutSettings.for_site(context['request'].site).logo:
+        layout_settings = LayoutSettings.for_site(context['request'].site)
+        relative_path = layout_settings.logo.get_rendition('original').url
+    return site_url + relative_path
```

Comment:
`relative_path` is not necessarily defined if none of the if/else statements are ever triggered. In the case where there is no og_image, no cover_image, and no logo.

In that case, you should probably just return an empty string.